compute_sdk/tests/unit/test_mpi_functions.py

```diff
@@ -0,0 +1,18 @@
+import os
+
+import pytest
+from globus_compute_sdk.sdk.mpi_function import MPIFunction
+
+
+@pytest.fixture
+def run_in_tmp_dir(tmp_path):
+    os.chdir(tmp_path)
+    return tmp_path
```

Comment:
Please use the fixture to also _undo_ the action.

1. pre-test setup
1. _actual test_
1. post-test cleanup

Meanwhile, for the test, is the `run_in_tmp_dir` necessary?  What purpose does it provide here, as nominally `hostname` only emits to stdout?

Reply:
These functions all drop temp files which do not get cleaned up, which is why I want to drop them in a `tmp_dir`.
I've updated the fixture to pop back into the original cwd, but there's no effort to delete the dir explicitly since `tmp_path` auto removes these.

Reply:
Ah; I didn't realize they dropped temp files.  :+1: